eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -311,4 +271,70 @@ public function extractFilesFromXml($xml)
 
         return null;
     }
+
+    /**
+     * Checks whether image reference for given version already exists.
+     *
+     * @param int $fieldId
+     * @param \eZ\Publish\SPI\Persistence\Content\VersionInfo $versionInfo
+     *
+     * @return bool
+     */
+    private function imageReferenceExistsForVersion(int $fieldId, VersionInfo $versionInfo): bool
+    {
+        $selectQuery = $this->connection->createQueryBuilder();
+        $selectQuery
+            ->select($this->connection->quoteIdentifier('id'))
+            ->from($this->connection->quoteIdentifier('ezimagefile'))
+            ->where(
+                $selectQuery->expr()->andX(
+                    $selectQuery->expr()->eq(
+                        $this->connection->quoteIdentifier('contentobject_attribute_id'),
+                        ':fieldId'
+                    ),
+                    $selectQuery->expr()->eq(
+                        $this->connection->quoteIdentifier('version'),
+                        ':versionNo'
+                    )
+                )
+            )
+            ->setParameter(':fieldId', $fieldId, PDO::PARAM_INT)
+            ->setParameter(':versionNo', $versionInfo->versionNo, PDO::PARAM_INT);
+
+        $statement = $selectQuery->execute();
+
+        return (int) $statement->fetchColumn() !== 0;
```

Comment:
This gave me a mild headache. Please reformat it:
```suggestion
        return ((int)$statement->fetchColumn()) !== 0;
```
yes, I know, we all should understand priorities for all expressions, w/o parentheses. But I'm bad. Alternatively you could go Yoda style:
```suggestion
return 0 !== (int)$statement->fetchColumn();
```
but remember to keep casting close to a variable or an expression you're casting. No space.

_Yeah, I see now that we had something similar for `canRemoveImageReference`. Nevertheless, it's still confusing for me._

Reply:
> This gave me a mild headache.

Me too, no worries. I just wanted to keep the same convention. It's already changed in https://github.com/ezsystems/ezpublish-kernel/pull/2835/commits/2daa15d1853c2d6413b7ff77080ad9987044df56: